Numba 0.50.1 not lowering boolean8 to bool

INFO:

```
E       numba.core.errors.LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
E       No definition for lowering <class 'bool'>(boolean8,) -> bool
E
E       File "rbc/tests/test_omnisci_array.py", line 251:
E           def array_even_sum_int32(b, x):
E               <source elided>
E               for i in range(n):
E                   if b[i]:
E                   ^
E
E       During: lowering "$32pred = call bool32($30binary_subscr.4, func=bool32, args=(Var($30binary_subscr.4, test_omnisci_array.py:251),), kws=(), vararg=None)" at /work/guilhermel/git/rbc/rbc/tests/test_omnisci_array.py (251)

/home/guilhermel/miniconda3/envs/rbc-dev/lib/python3.8/site-packages/numba/core/utils.py:81: LoweringError
```

Steps to reproduce:
```bash
pytest -v -r s -x rbc --disable-warnings -k 'test_even_sum'
```

--- a/rbc/typesystem.py
+++ b/rbc/typesystem.py
@@ -1145,8 +1145,7 @@ if nb is not None:
     def literal_boolean_to_booleanN(context, builder, fromty, toty, val):
         llty = context.get_value_type(toty)
         return builder.zext(val, llty)
-    
-    
+
     @extending.lower_builtin(bool, Boolean8)
     def boolean8_to_bool(context, builder, sig, args):
         [val] = args

--- a/rbc/typesystem.py
+++ b/rbc/typesystem.py
@@ -11,10 +11,10 @@ from .utils import get_version
 try:
     import numba as nb
     if get_version('numba') >= (0, 49):
-        from numba.core import typing, datamodel
+        from numba.core import typing, datamodel, extending
         from numba.core.imputils import lower_cast
     else:
-        from numba import typing, datamodel
+        from numba import typing, datamodel, extending
         from numba.targets.imputils import lower_cast
     nb_NA_message = None
 except ImportError as msg:
@@ -1145,6 +1145,12 @@ if nb is not None:
     def literal_boolean_to_booleanN(context, builder, fromty, toty, val):
         llty = context.get_value_type(toty)
         return builder.zext(val, llty)
+    
+    
+    @extending.lower_builtin(bool, Boolean8)
+    def boolean8_to_bool(context, builder, sig, args):
+        [val] = args
+        return builder.icmp_signed('!=', val, val.type(0))
 
 
 def make_numba_struct(name, members, _cache={}):

